core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -721,6 +715,15 @@ object Doc {
   def split(str: String, pat: Regex = Doc.splitWhitespace, sep: Doc = Doc.lineOrSpace): Doc =
     foldDocs(pat.pattern.split(str, -1).map(Doc.text))((x, y) => x + (sep + y))
 
+  /**
+   * Render whenFlat if the document (or enclosing document) is grouped.
+   * Otherwise render default.
+   *
+   * Invariant: width(default) <= width(whenFlat)
+   */
+  def flatAlt(default: Doc, whenFlat: Doc): Doc =
```

Comment:
I don't like making this public. So far, our API has no unsafe calls. This changes it since users must maintain the invariant.

Reply:
Yeah, I didn't know whether to do so or not.  I was thinking that it allowed more fine grained control over custom rendering, but indeed, the docs can be totally unrelated.  

Currently users can add Text blocks with newlines and spaces, which is technically unsafe.  I agree that this is more extreme.  I am happy to make it private and add it back if it's ever needed in Scalafmt.

Reply:
My last comment stemmed from a basic misunderstanding of the API.  Will drop the `flatAlt` function.